sympy/printing/latex.py

```diff
@@ -1813,7 +1813,50 @@ def _print_PolynomialRingBase(self, expr):
 
     def _print_Poly(self, poly):
         cls = poly.__class__.__name__
-        expr = self._print(poly.as_expr())
+        terms, gens = [], [s for s in poly.gens]
```

Comment:
These would be more readable if on separate lines. 
Also, why `[s for s in poly.gens]` instead of `list(poly.gens)`?
Also, why do this at all? (introduce variable gens, convert poly.gens to a list)?  `poly.gens` is a tuple which will work just fine for your purposes: `gens[i]` becomes `poly.gens[i]`.

Reply:
Also, a variable named `gens`  already appears near the end of the function, where it means something different. Variable names shouldn't be reused like this, it makes the code harder to read.